forc/src/ops/forc_init.rs

```diff
@@ -34,29 +34,38 @@ fn print_welcome_message() {
     );
 }
 
-pub fn canonicalize(path: &PathBuf) -> Result<String> {
-    match std::fs::canonicalize(path) {
-        Ok(p) => {
-            let as_str = String::from(p.to_string_lossy());
-            Ok(as_str)
-        }
-        Err(e) => anyhow::bail!("Could not derive canonical path: '{}'", e),
-    }
-}
-
 pub fn init(command: InitCommand) -> Result<()> {
     let project_dir = match &command.path {
         Some(p) => PathBuf::from(p),
-        None => std::env::current_dir().unwrap(),
+        None => {
+            std::env::current_dir().context("Failed to get current directory for forc init.")?
+        }
     };
 
+    if !project_dir.is_dir() {
+        anyhow::bail!("'{}' is not a valid directory.", project_dir.display());
+    }
+
+    if Path::exists(&Path::new(&project_dir).join("Forc.toml")) {
+        anyhow::bail!(
+            "'{}' already includes a Forc.toml file.",
+            project_dir.display()
+        );
+    }
+
+    if command.verbose {
+        println_light_blue(&format!(
+            "\nUsing project directory at {}",
+            project_dir.canonicalize()?.display()
+        ))
+    }
+
     let project_name = command.name.unwrap_or_else(|| {
         project_dir
-            .to_str()
-            .unwrap()
-            .split('/')
-            .last()
+            .file_stem()
+            .context("Failed to infer project name from directory name.")
             .unwrap()
+            .to_string_lossy()
             .to_string()
     });
```

Comment:
We should return the error here rather than panicking, e.g. something like this:

```rust
let project_name = command.name.ok_or_else(|| {
    let dir_name = project_dir
        .file_stem()
        .context("Failed to infer project name from directory name.")?
        .to_string_lossy();
    Ok(dir_name)
});